Marlin/src/pins/pins_postprocess.h

```diff
@@ -564,19 +564,29 @@
       #elif Z_HOME_TO_MAX && defined(Z_MIN_PIN)
         #define Z2_STOP_PIN Z_MIN_PIN
       #else
-        #error"Cannot auto allocate Z2_STOP_PIN. Define Z2_STOP_PIN in Configuration_adv.h"
+        #error "Cannot auto allocate Z2_STOP_PIN. Define Z2_STOP_PIN in Configuration_adv.h"
       #endif
     #endif
-
     #if NUM_Z_STEPPERS >= 3
       #if ((Z_HOME_TO_MIN && !defined(Z3_MIN_PIN)) || (Z_HOME_TO_MAX && !defined(Z3_MAX_PIN))) && !defined(Z3_STOP_PIN)
-        #error"Cannot auto allocate Z3_STOP_PIN. Define Z3_STOP_PIN in Configuration_adv.h"
+        #if Z_HOME_TO_MIN && defined(Z_MAX_PIN)
+          #define Z3_STOP_PIN Z_MAX_PIN
+        #elif Z_HOME_TO_MAX && defined(Z_MIN_PIN)
+          #define Z3_STOP_PIN Z_MIN_PIN
```

Comment:
This can be in conflict. If the user has not defined Z2_STOP_PIN, then Z_MIN_PIN is already allocated to  Z2_STOP_PIN and cannot be used for Z3_STOP_PIN.